CHANGES

```diff
@@ -1,3 +1,7 @@
+20190710 code: add instchown to set ownership and mode on installed files.
+20190709 code: add instpackage to install without root access.
+20190710 code: removed idedit.
```

Comment:
I'd either (1) keep it for now, or (2) remove the whole horse it rode in on.

Argument for keep-for-now: conservatism. "Delete unused code" is generally [slotted for later](https://github.com/notqmail/notqmail/wiki/Release-Plans#maybe-later). If the `idedit` code were broken by your changes, the chance to avoid paying maintenance on unwanted code would make a compelling exception. But it's merely as useless as before, not broken.

Argument for entire-horse: When we claim "Being easily packaged by OS integrators" is a goal, none of us means [DJB's var-qmail packages](https://cr.yp.to/qmail/var-qmail.html). As a successor to qmail, we will need to send clear signals to OS integrators about what we do mean. One very clear signal: our first release is entirely dedicated to providing defaults that allow building as non-root, installing to a staging area, and producing FHS-compatible binary packages. A second clear signal would be to delete everything else related to DJB's var-qmail package scheme, which I believe amounts to these:

- `BIN.Makefile`
- `BIN.README`
- `install-big`
- `config-fast`

Reply:
I'll pull the commit removing idedit out of this series and bundle it with a new PR removing all of the var-qmail package infrastructure.

config-fast is unusual in your list here.  Gerrit Pape installs that script and runs it in postinst in the netqmail Debian package.  Removing it will break that package.  The ultimate result of our packaging effort will let us broadly remove code from the debian package, as those features will essentially move in-tree.  I'd like to see our packaging work release-ready before making a netqmail package breaking change--so we can make good on that promise of code removal.

As a cross check to the above, the file INSTALL.ctl mentions both config and config-fast.  Removing config-fast.sh naturally brings up the request of also removing config.sh.  config.sh is only available in-tree, uses several supplementary programs (DNS queries), and represents work that needs to be done at install time.  It is certainly true that we support building and installing on the same host, but it is also possible that this use-case will be fully accommodated by the package build pipeline.  This confirms that removing config-fast.sh requires more caution (is a larger change) than idedit or install-big.

I'll submit a patch to remove install-big and idedit (the core of the var-qmail package support) but decline to remove config-fast in favor of deferring that decision to a later PR that better accounts for (at least) Debian packaging and changes to INSTALL.ctl and possibly config.sh.

Whole horse (2), but for config-fast (1).  With one extra bonus removal, stay tuned.  ;-)

Reply:
Forgot to mention, by way of +1: I use `config-fast` in pkgsrc too.